native/src/main/java/io/ballerina/stdlib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,350 @@
+        String[] parts = configValue.split(SKIP_LINE_RANGE_SEP);
+        if (parts.length != 2) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+        }
+        try {
+            int start = Integer.parseInt(parts[0]);
+            int end = Integer.parseInt(parts[1]);
+            int size = end - start + 1;
+            if (size <= 0) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_RANGE_FOR_SKIPLINES);
+            }
+            long[] result = new long[size];
+            for (int i = 0; i < size; i++) {
+                result[i] = start + i;
+            }
+            return result;
+        } catch (NumberFormatException e) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_VALUE_FOR_SKIPLINES);
+        }
+    }
+
+    public static long[] getSkipDataRows(Object skipLines) {
+        long[] skipDataRows;
+        if (skipLines == null) {
+            return new long[]{};
+        }
+        if (skipLines instanceof BArray) {
+            BArray skipLinesArray = (BArray) skipLines;
+            if (skipLinesArray.getLength() == 0) {
+                return new long[]{};
+            }
+            skipDataRows = (skipLinesArray).getIntArray();
+            return skipDataRows;
+        }
+
+        skipDataRows = getSkipLinesFromStringConfigValue(StringUtils.getStringValue(skipLines));
+        return skipDataRows;
+    }
+
+    public static boolean isNullValue(Object nullValue, Object value) {
+        if (value == null) {
+            return true;
+        }
+        if (value instanceof BString) {
+            value = StringUtils.getStringValue(value);
+        }
+        if (value instanceof String v) {
+            if ((nullValue == null) && (Constants.Values.NULL.equalsIgnoreCase(v)
+                    || Constants.Values.BALLERINA_NULL.equalsIgnoreCase(v))) {
+                return true;
+            }
+            if (nullValue != null && value.equals(StringUtils.getStringValue(nullValue))) {
+                return true;
+            }
+            return false;
+        }
+        return false;
+    }
+
+    public static boolean isCharContainsInLineTerminatorUserConfig(char c, Object lineTerminatorObj) {
+        if (lineTerminatorObj instanceof BArray) {
+            Object[] lineTerminators = ((BArray) lineTerminatorObj).getValues();
+            for (Object lineTerminator: lineTerminators) {
+                if (lineTerminator != null && c == Constants.LineTerminator.LF) {
+                    String lineTerminatorString = lineTerminator.toString();
+                    if (isCarriageTokenPresent) {
+                        if (lineTerminatorString.equals(Constants.LineTerminator.CRLF)) {
+                            return true;
+                        }
+                        continue;
+                    }
+                    return true;
+                }
```

Comment:
I feel like this is somewhat too deeply nested, shall we factor this to a function that returns a Optional<Boolean> and use early returns in that to de-nest the method. You can return empty to indicated continue